uaclient/config.py

```diff
@@ -160,14 +161,29 @@ def data_path(self, key=None, private=True):
             return os.path.join(data_dir, self.data_paths[key])
         return os.path.join(data_dir, key)
 
+    def delete_cache_key(self, key):
+        """Remove specific cache file."""
+        if not key:
+            return
+        if key.startswith('machine-access'):
+            self._entitlements = None
+        elif key == 'account-contracts':
+            self._contracts = None
+        elif key == 'machine-token':
+            self._machine_token = None
+        for private in (True, False):
+            cache_path = self.data_path(key, private)
+            if os.path.exists(cache_path):
+                os.unlink(cache_path)
+
     def delete_cache(self):
-        """Remove all configuration cached response files class attributes."""
+        """Remove configuration cached response files class attributes."""
         self._contracts = None
         self._entitlements = None
         self._machine_token = None
-        for key in self.data_paths.keys():
+        for path_key in self.data_paths.keys():
             for private in (True, False):
```

Comment:
Should we just be calling `self.delete_cache_key` in this for loop?

Reply:
Good point. we should be calling delete_cache_key to avoid duplication.